cranelift/codegen/src/isa/aarch64/lower/isle.rs

```diff
@@ -0,0 +1,192 @@
+//! ISLE integration glue code for aarch64 lowering.
+
+// Pull in the ISLE generated code.
+pub mod generated_code;
+
+// Types that the generated ISLE code uses via `use super::*`.
+use super::{
+    zero_reg, AMode, ASIMDFPModImm, ASIMDMovModImm, AtomicRmwOp, BranchTarget, CallIndInfo,
+    CallInfo, Cond, CondBrKind, ExtendOp, FPUOpRI, Imm12, ImmLogic, ImmShift, Inst as MInst,
+    JTSequenceInfo, MachLabel, MoveWideConst, Opcode, OperandSize, PairAMode, Reg, ScalarSize,
+    ShiftOpAndAmt, UImm5, VectorSize, Writable, NZCV,
+};
+use crate::isa::aarch64::settings as aarch64_settings;
+use crate::{
+    binemit::CodeOffset,
+    ir::{
+        immediates::*, types::*, ExternalName, Inst, InstructionData, MemFlags, TrapCode, Value,
+        ValueLabel, ValueList,
+    },
+    isa::aarch64::inst::aarch64_map_regs,
+    isa::unwind::UnwindInst,
+    machinst::{get_output_reg, InsnOutput, LowerCtx, RegRenamer},
+};
+use smallvec::SmallVec;
+use std::boxed::Box;
+use std::vec::Vec;
+
+type Unit = ();
```

Comment:
Some of these type-defs seem like they could be shared with x64 than `pub use common::*` into this module, probably?